pkg/api/customization/catalog/catalog.go

```diff
@@ -99,3 +100,34 @@ func (a ActionHandler) ExportYamlHandler(apiContext *types.APIContext, next type
 	}
 	return nil
 }
+
+func (a ActionHandler) RefreshProjectCatalogActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	if actionName != "refresh" {
+		return httperror.NewAPIError(httperror.NotFound, "not found")
+	}
+
+	prjCatalogs := []v3.ProjectCatalog{}
+	if apiContext.ID != "" {
+		catalog, err := a.ProjectCatalogClient.Get(apiContext.ID, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		prjCatalogs = append(prjCatalogs, *catalog)
+	} else {
+		catalogList, err := a.ProjectCatalogClient.List(metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		for _, catalog := range catalogList.Items {
+			prjCatalogs = append(prjCatalogs, catalog)
+		}
+	}
+	for _, catalog := range prjCatalogs {
+		catalog.Status.LastRefreshTimestamp = time.Now().Format(time.RFC3339)
+		v3.CatalogConditionRefreshed.Unknown(&catalog)
+		if _, err := a.ProjectCatalogClient.Update(&catalog); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
What does the "refresh" action do? I se we are updating the "LastRefreshTimestamp" here and saving the CRD but what does really do the refresh of the templates? (I assume by refresh we mean re-fetch all the templates?)

Reply:
Also what does v3.CatalogConditionRefreshed.Unknown(&catalog) do?

Reply:
Discussed offline that this will eventually call the sync method

Reply:
discussed offline